lib/_http_outgoing.js

```diff
@@ -109,35 +109,29 @@ OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 // any messages, before ever calling this.  In that case, just skip
 // it, since something else is destroying this connection anyway.
 OutgoingMessage.prototype.destroy = function(error) {
-  if (this.socket)
+  if (this.socket) {
+    // If the socket is corked from a write, uncork it before destroying it.
+    if (this.socket._writableState.corked)
+      this.socket.uncork();
```

Comment:
I'm uncorking on destroy because, without it, it broke a test — leaving it corked would be backwards-incompatible as well. Unfortunately this is part of a gray area. There isn't any *guarantee* that the message would be flushed regardless, as `net.js` may not be able to write it synchronously. If the message is buffered to be written asynchronously, and `res.destroy()` is called before that happens, the message will not be written at all. It's an odd gray area, but this is the best way to keep code breakage down.